ui/pages/asset/components/asset-page.tsx

```diff
@@ -221,28 +243,52 @@ const AssetPage = ({
               <Text variant={TextVariant.headingMd} paddingBottom={4}>
                 {t('tokenDetails')}
               </Text>
-              {renderRow(
-                t('contractAddress'),
-                <AddressCopyButton address={address} shorten />,
-              )}
               <Box
                 display={Display.Flex}
                 flexDirection={FlexDirection.Column}
                 gap={2}
               >
-                {asset.decimals !== undefined &&
-                  renderRow(t('tokenDecimal'), <Text>{asset.decimals}</Text>)}
-                {asset.aggregators && asset.aggregators?.length > 0 && (
+                {type === AssetType.token && (
                   <Box>
-                    <Text
-                      color={TextColor.textAlternative}
-                      variant={TextVariant.bodyMdMedium}
+                    {renderRow(
+                      t('contractAddress'),
+                      <AddressCopyButton address={address} shorten />,
+                    )}
+                    <Box
+                      display={Display.Flex}
+                      flexDirection={FlexDirection.Column}
+                      gap={2}
                     >
-                      {t('tokenList')}
-                    </Text>
-                    <Text>{asset.aggregators?.join(', ')}</Text>
+                      {asset.decimals !== undefined &&
+                        renderRow(
+                          t('tokenDecimal'),
+                          <Text>{asset.decimals}</Text>,
+                        )}
+                      {asset.aggregators && asset.aggregators?.length > 0 && (
+                        <Box>
+                          <Text
+                            color={TextColor.textAlternative}
+                            variant={TextVariant.bodyMdMedium}
+                          >
+                            {t('tokenList')}
+                          </Text>
+                          <Text>{asset.aggregators?.join(', ')}</Text>
+                        </Box>
+                      )}
+                    </Box>
                   </Box>
                 )}
+                {renderRow(
+                  t('spendingCaps'),
+                  <a
```

Comment:
I think `<Text as ="a" .... color={TextColor.PrimaryDefault}` is what the Design System team would prefer.  cc; @georgewrmarshall

Reply:
The CSS class `asset-page__spending-caps` seems to have a lot more than just color={TextColor.PrimaryDefault}. It has hover styles, as well as text decoration styles. Would the change you suggested also entail removing the CSS class? Otherwise why add color={TextColor.primaryDefault} if we need the CSS anyways?

Thanks! I'm just starting to work on the extension so I'm trying to learn from you.

Reply:
Hey @ziad-saab, ideally we want to use *component-library* components and style utility props wherever possible. This will reduce the amount of CSS needed during loading and improve UI consistency across the extension.

Reply:
@georgewrmarshall thank you! So it's fine to use a mix of style props and CSS, correct?

Reply:
Yes, unfortunately the style props are limited so you will have to do some things with CSS so using the style props and then adding a classname to add additional styles is fine!